### PR TITLE
Display metadata hash if found

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/app/renderer.js [hunk 1]
```diff
@@ -91,6 +91,13 @@ Renderer.prototype.contracts = function (data, source) {
     })
   }
 
+  var retrieveMetadataHash = function (bytecode) {
+    var match = /a165627a7a72305820([0-9a-f]{64})0029$/.exec(bytecode)
```
Comment (chriseth) [inline src/app/renderer.js]: case-insensitive regex to be more flexible?

File: src/app/renderer.js [hunk 2]
```diff
@@ -101,6 +108,12 @@ Renderer.prototype.contracts = function (data, source) {
 
     if (contract.bytecode) {
       $contractOutput.append(uiHelper.preRow('Web3 deploy', uiHelper.gethDeploy(contractName.toLowerCase(), contract['interface'], contract.bytecode), 'deploy'))
+
+      // check if there's a metadata hash appended
+      var metadataHash = retrieveMetadataHash(contract.bytecode)
+      if (metadataHash) {
+        $contractOutput.append(uiHelper.tableRow('Metadata location', 'bzzr://' + metadataHash))
```
Comment (chriseth) [inline src/app/renderer.js]: We could use `'http://swarm-gateways.net/bzzr:/' + metadataHash` until we have support for the bzzr protocol

File: src/app/renderer.js (hunk 1 above)
Comment (axic) [inline src/app/renderer.js]: I think we should rather just lowercase the bytecode so the output is consistent.

File: src/app/renderer.js (hunk 2 above)
Comment (axic) [inline src/app/renderer.js]: Since we're not solving the upload process yet I thought just to display the actual metadata and the URL.

**Pull request merged**